pgpy/pgp.py

```diff
@@ -1374,6 +1374,15 @@ def signers(self):
         return {sig.signer for sig in self.__sig__}
 
     @property
+    def revocation_signatures(self):
+        keyid, keytype = (self.fingerprint.keyid, SignatureType.KeyRevocation) if self.is_primary \
+            else (self.parent.fingerprint.keyid, SignatureType.SubkeyRevocation)
+
+        for sig in iter(sig for sig in self._signatures
+                        if all([sig.type == keytype, sig.signer == keyid, not sig.is_expired])):
```

Comment:
i'm not sure `sig.signer == keyid` is sufficient here. we should actually verify the signatures directly, but i am not quite sure how to do that.

Reply:
Well, that comparison is there only to filter the signatures. It is up to the user of PGPy to verify them, similar to how signatures from `self_signatures` and so are not verified by PGPy itself.

Reply:
i see. then i think this should be made very explicit in the documentation, because that could lead to serious security vulnerabilities in the API's consumers.

Reply:
unfortunately, we don't have a choice with v4 signatures. sig.signer references the [Issuer subpacket](https://tools.ietf.org/html/rfc4880#section-5.2.3.5) on the signature, which only contains the 8 byte keyid of the signing key.

PGPy doesn't insist on verifying self signatures because:

a) that can introduce a lot of overhead for users who may not need to re-verify their keys every time they load one up

b) at what point during a key's lifecycle would self-signature verification be introduced? It can't always be immediately after the object exists, because they won't always have one yet, like if it's being generated, etc.

along similar lines here, we can't depend on being able to verify those revocation signatures because they aren't guaranteed to be self-signatures, and a key has no way to obtain a reference to another key (the revoker's key may not even be anywhere in memory). So it's probably more universally useful to provide a set of candidates and let the consumer sort out verifying them if needed, as long as doing so is convenient enough.

Reply:
makes sense then, i retract my objection. :)